tests/test_unique_name.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+from tk_toolchain.testing import create_unique_name
+
+
+def test_with_env_var():
+    """
+    Ensure the env var is added to the name
+    """
+    os.environ["SHOTGUN_TEST_ENTITY_SUFFIX"] = "Potatoe"
```

Comment:
There's a small problem with your tests here. You're updating the environment variable, which means you've now modified the environment and that state is going to leak into other unit tests. In general, we can to avoid that sort of practice. To avoid global state spilling into other tests, we usually use `monkeypatch` when using pytest. Here's a link to the [documentation](https://docs.pytest.org/en/stable/monkeypatch.html), and here's [a link](https://docs.pytest.org/en/stable/monkeypatch.html#monkeypatching-environment-variables) to the section specific to setting/deleting environment variables.

In your case, it would be more appropriate if you did this

```
def test_with_env_var(monkeypatch):
    monkeypatch.setenv("SHOTGUN_TEST_ENTITY_SUFFIX", "Potatoe")
    ...
```

This will ensure the environment variable gets set during the test and reset after it.